rpc/filters.go

```diff
@@ -12,32 +13,80 @@ import (
 	Used to set the criteria passed in from RPC params
 */
 
+// Backend implements the functionality needed to filter changes.
+// Implemented by PublicEthAPI.
+type Backend interface {
+	BlockNumber() (hexutil.Uint64, error)
+	GetBlockByNumber(blockNum BlockNumber, fullTx bool) (map[string]interface{}, error)
+	PendingTransactions() ([]*Transaction, error)
+	GetTxLogs(txHash common.Hash) ([]*ethtypes.Log, error)
+	// Bloom methods
+}
+
 // Filter can be used to retrieve and filter logs.
 type Filter struct {
-	addresses []common.Address
-	topics    [][]common.Hash
-
-	block common.Hash // Block hash if filtering a single block
+	backend            Backend
+	fromBlock, toBlock *big.Int         // start and end block numbers
```

Comment:
probably will be easier to reference by i64, Geth and Tendermint reference block numbers by that. I don't see why this would need to be a big int?

Reply:
will change

Reply:
nvm, it's cause geth uses *big.Int https://godoc.org/github.com/ethereum/go-ethereum#FilterQuery

Reply:
`FilterQuery` and this type are different, can keep this if you'd like, but you're just going to have to do manual conversions later I think

Reply:
oh, which FilterQuery are you referring to?

Reply:
not one in specific, was referring to geth one, but confusion may be because this is the type to be used for the request through the API and this filter type you created here is the actual filter used to filter logs (they are not the same type)